### DISCUSSION TITLE
Can diagrams.net use with Microsoft OneDrive for Business' Sharepoint?

### DISCUSSION THREAD

**Discussion by noissefnoc** [jgraph/drawio#2742] [Q&A] [answered]
Hello,
I've just started using diagrams.net in Microsoft 365(Enterprise license).
I can save diagrams on OneDrive space but I can't save on Sharepoint space (some Sharepoint spaces name displays) with error 'Invalid selection'.

I ask my organization's administrator and they wants me to check whether diagrams.net supports Microsoft OneDrive for Business (not consumer one) or not.

I see diagrams.net top page's "Sharepoint and OneDrive" section so I think diagrams.net supports Sharepoint for all plan.

Could anyone tell me whether diagrams.net supports Microsoft OneDrive for Business (not consumer one) or not?

Thanks for the help.

**Comment by davidjgraph** [comment, marked as answer]
It does support OD for business, but it's possible we're not detecting it's the business API that needs to be used. What does the URL called for authenication look like (you can remove company names)?

**Reply by noissefnoc** [reply]
Thanks for your quick answer. I'll ask my organization's administrator and reply to this thread.

**Reply by noissefnoc** [reply]
I got answer from my organization's administrator. They says following

> https://login.microsoftonline.com/<TENANT_ID>/oauth2/v2.0/authorize

I think their answer might not be your expect. (I expected diagrams.net app's auth URL.). If this URL doesn't match your expect, please tell me where to get URL.


Additionally, I ask they what scope permitted to diagrams.net app. They says it has `Sites.Read.All` but does not have `Site.Write` so this is because I can't save diagrams to Sharepoint.

**Reply by noissefnoc** [reply]
@davidjgraph Could you check organization's administrator's answer?

**Reply by m-mohamedin** [reply]
Can you explain more (where in the Sharepoint sites you want to store) and with screenshots if possible.
Is your account in the form username@company.onmicrosoft.com?
This is what we see in a business account:
![Screen Shot 2022-04-14 at 5 53 11 PM](https://user-images.githubusercontent.com/24944765/163482717-bdee3d6f-42b7-4e17-9614-9fec31499712.png)

**Reply by noissefnoc** [reply]
@ashrafteleb I'm sorry for replying late. (I couldn't get up from getting side effects from COVID-19 vaccine.)

I realized my mistake from your capture. I tried to save diagrams.net files under `Sherepoint Sites > SITE_NAME` ( not under `Sherepoint Sites > SITE_NAME > Documents` ).

So now I can use diagrams.net with Microsoft OneDrive for Business' Sharepoint.

Thanks for your help.
